Add option to disallow fallback for certain packages

neuronist opened the pull request:
This is my hacky attempt at providing the functionality requested on #214.

Let me know if you have any suggestions or if you don't think this is the right approach to tackle this problem.

This is useful if your internal packages shadow names from Pypi and you want to avoid your internal packages getting upgraded to a higher version of a different package present in Pypi.
To use this you just have to add a list of package names in config.ini. E.g. 
```
pypi.disallow_fallback =
  flask
  numpy
```

mamoit commented:
You have broken some tests

stevearc commented on pypicloud/access/base.py:
```diff
@@ -129,6 +132,11 @@ def allowed_permissions(self, package):
+        # add fallback permissions
+        for principal in all_perms:
+            if package not in self.disallow_fallback:
+                all_perms[principal] += ("fallback",)
```
Slight preference to put the `if` before the `for` to avoid repetitive lookups in the array.

neuronist replied:
fixed with 0bc76bb

neuronist commented:
Thanks for the reply and feedback! I will try to formalize this PR this week. A bit loaded with work currently.

neuronist commented:
@stevearc hey man I changed the for loop, fixed the tests, added 2 new ones and also added relevant configuration documentation. :)

I'm a bit confused with the tests and where to put them or if the ones I made are ok. Please give me some feedback to help me navigate your test setup if necessary. :)

Sorry for the number of different commits. :s  
I tried setting up the test environment locally but wasn't able too and was losing too much time on it.

stevearc commented:
Thanks for the PR! Merged into the [1.0 branch](https://github.com/stevearc/pypicloud/tree/1.0)

neuronist commented:
Thanks! I'll close this PR then.
